src/core/search/indices.cc

```diff
@@ -214,11 +214,13 @@ struct HnswlibAdapter {
     world_.markDelete(id);
   }
 
-  vector<pair<float, DocId>> Knn(float* target, size_t k) {
+  vector<pair<float, DocId>> Knn(float* target, size_t k, std::optional<size_t> ef) {
+    world_.setEf(ef.value_or(10));
```

Comment:
please add a constant `DEFAULT_EF_CONSTRUCT` in the header class and add a comment around `Knn` declaration that it uses `DEFAULT_EF_CONSTRUCT` by default

Reply:
Why so complicated, there is one place that interferes with hnswlib and the closest place to put it is there